### ISSUE TITLE
RFE: support for no_device in aws_instance ephemeral_block_device

### ISSUE THREAD

**Issue by mblakele** [hashicorp/terraform#8455]
### Terraform Version

Terraform v0.7.1
### Affected Resource(s)

Please list the resources as a list, for example:
- aws_instance
### Terraform Configuration Files

``` hcl
resource "aws_instance" "test" {
  # Block ephemeral storage, which confuses the setup process.
  ephemeral_block_device {
    device_name = "/dev/sdb"
    no_device = "true"
    virtual_name = "ephemeral0"
  }
}
```
### Expected Behavior

If I'm reading https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/block-device-mapping-concepts.html correctly, terraform should set `NoDevice` for the specified `device_name`. 

```
To add volumes to an instance using the command line
...

For example, suppose that an EBS-backed AMI specifies the following block device mapping:

/dev/sdb=ephemeral0
/dev/sdh=snap-1234567890abcdef0
/dev/sdj=:100
To prevent /dev/sdj from attaching to an instance launched from this AMI, use the following mapping:

{
    "DeviceName": "/dev/sdj",
    "NoDevice": ""
}
```

Also `virtual_name` probably shouldn't be there when `no_device` is set — but terraform 0.7.1 requires it.
### Actual Behavior

```
aws_instance.test: ephemeral_block_device.0: invalid or unknown key: no_device
```
### Steps to Reproduce
1. `terraform plan`

Possible workaround: create the AMI without the ephemeral mappings. But I'd rather not do that because I'd have to maintain two AMIs and that would be the only difference between them.

**Comment by jweir** [comment]
This issue is burning me too.  Packer had this issue and was fixed via https://github.com/mitchellh/packer/pull/2672/commits/2a94b596ce93237517026ddd220da8b12f0132c7

**Comment by timonwong** [comment]
Hi, when this issue is fixed, it should address #5390, when the BlockingMappingDevice from an AMI contains ephemerals, "NoDevice" have to be set on EBS-only instance, or the AutoRecover feature from AWS won't work.
